x-pack/test/saved_object_api_integration/common/suites/bulk_resolve.ts

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { SuperTest } from 'supertest';
+import { TEST_CASES } from './resolve';
+import { SPACES } from '../lib/spaces';
+import {
+  createRequest,
+  expectResponses,
+  getUrlPrefix,
+  getTestTitle,
+} from '../lib/saved_object_test_utils';
+import { ExpectResponseBody, TestCase, TestDefinition, TestSuite } from '../lib/types';
+
+export interface BulkResolveTestDefinition extends TestDefinition {
+  request: Array<{ type: string; id: string }>;
+}
+export type BulkResolveTestSuite = TestSuite<BulkResolveTestDefinition>;
+export interface BulkResolveTestCase extends TestCase {
+  expectedOutcome?: 'exactMatch' | 'aliasMatch' | 'conflict';
+  expectedId?: string;
+  expectedAliasTargetId?: string;
+}
+
+export { TEST_CASES }; // re-export the (non-bulk) resolve test cases
+
+export function bulkResolveTestSuiteFactory(esArchiver: any, supertest: SuperTest<any>) {
+  const expectSavedObjectForbidden = expectResponses.forbiddenTypes('bulk_get');
+  const expectResponseBody = (
+    testCases: BulkResolveTestCase | BulkResolveTestCase[],
+    statusCode: 200 | 403
+  ): ExpectResponseBody => async (response: Record<string, any>) => {
+    const testCaseArray = Array.isArray(testCases) ? testCases : [testCases];
+    if (statusCode === 403) {
+      const types = testCaseArray.map((x) => x.type);
+      await expectSavedObjectForbidden(types)(response);
+    } else {
+      // permitted
+      const resolvedObjects = response.body.resolved_objects;
+      expect(resolvedObjects).length(testCaseArray.length);
+      for (let i = 0; i < resolvedObjects.length; i++) {
+        const resolvedObject = resolvedObjects[i];
+        const testCase = testCaseArray[i];
+        const { expectedId: id, expectedOutcome, expectedAliasTargetId } = testCase;
+        await expectResponses.permitted(resolvedObject.saved_object, {
+          ...testCase,
+          ...(!testCase.failure && id && { id }), // use expected ID instead of the requested ID iff the case was *not* a failure
+        });
+        if (!testCase.failure) {
+          expect(resolvedObject.outcome).to.eql(expectedOutcome);
+          if (expectedOutcome === 'conflict' || expectedOutcome === 'aliasMatch') {
+            expect(resolvedObject.alias_target_id).to.eql(expectedAliasTargetId);
+          } else {
+            expect(resolvedObject.alias_target_id).to.eql(undefined);
+          }
+        }
+        // TODO: add assertions for redacted namespaces (this already exists in the bulkCreate test suite)
```

Comment:
Is this something we want to do in this PR, or should we open an issue to track this work? If the latter, can we add a link to the issue here?

Reply:
Good point, I'll open an issue for follow-on work and link it here.

Reply:
Created an issue: #112455